Angular/angular-tour-of-heroes/src/app/components/hero-detail/hero-detail.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component, EventEmitter, Input, OnChanges, Output, SimpleChanges } from '@angular/core';
+import { Hero } from 'src/app/models/hero';
+import { HeroService } from 'src/app/services/hero.service';
+
+class HeroDetail {
+  id?: number = undefined;
+  name: string = '';
+}
+
+@Component({
+  selector: 'app-hero-detail',
+  templateUrl: './hero-detail.component.html',
+  styleUrls: ['./hero-detail.component.scss']
```

Comment:
You'd want to use `OnPush` change detection. Otherwise in the change detection cycle Angular would check the whole application for any changes. Interesting read on how change detection works with zone.js. Although with signals this will be a non issue.